promotion/drain.go

```diff
@@ -242,8 +242,26 @@ func (service *Service) RedeemAndTransferFunds(ctx context.Context, credentials
 	} else if *wallet.UserDepositAccountProvider == "bitflyer" {
 
 		transferID := uuid.NewV4().String()
+
 		totalF64, _ := total.Float64()
 
+		// get quote, make sure we dont go over 100K JPY
+		quote, err := service.bfClient.FetchQuote(ctx, "BAT_JPY", false)
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch bitflyer quote")
+		}
+
+		JPYLimit := decimal.NewFromFloat(100000.0)
+		var overLimitErr error
+
+		totalJPYTransfer := total.Mul(quote.Rate)
+
+		if totalJPYTransfer.GreaterThan(JPYLimit) {
+			over := JPYLimit.Sub(totalJPYTransfer).String()
+			totalF64, _ = JPYLimit.Div(quote.Rate).Floor().Float64()
+			overLimitErr = fmt.Errorf("transfer is over 100K JPY by %s", over)
```

Comment:
lets add the rate to this error string as well

Reply:
resolved in https://github.com/brave-intl/bat-go/pull/727/commits/db8854558cbde9dee599f3a5c0df5995b1ec42da